scripts/customize_conda.sh

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#            http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euxo pipefail
+
+# This customization-script can be used to customize the conda environment.
+# It expects the following metadata:
+#   conda-component: (Required) Must be either ANACONDA or MINICONDA3.
+#   conda-env-config-uri: (Optional) Must be a gsutil path to the yaml config
+#   file.
+#   conda-packages: (Optional) A list of conda packages with versions to be
+#   installed in the base environment. Must be of the format
+#   <pkg1>:<version1>_<pkg2>:<version2>...
+#   Example: conda-packages=pytorch:1.4.0_visions:0.7.1
+#   pip-packages: (Optional) A list of pip packages with versions to be
+#   installed in the base environment. Must be of the format
+#   <pkg1>:<version1>_<pkg2>:<version2>...
+#   Example: tokenizers:0.10.1_datasets:1.5.0
+# conda-env-config-uri is mutually exclusive with conda-packages and
+# pip-packages.
+
+
+function customize_conda() {
+  local -r conda_component=$(/usr/share/google/get_metadata_value attributes/conda-component)
```

Comment:
`local -r conda_component=$(/usr/share/google/get_metadata_value attributes/conda-component)` needs to be 2 lines (Google3 has this check), otherwise the error in `$(...)` will be ignored:

```
local conda_component
conda_component=$(/usr/share/google/get_metadata_value attributes/conda-component)
```

also other places.

Reply:
I have seen multiple uses like this in Google3 - https://source.corp.google.com/piper///depot/google3/cloud/hadoop/services/images/bdutil/components/activate/kerberos.sh;l=62?q=%22local%20-r%22&ss=piper%2FGoogle%2FPiper:google3%2Fcloud%2Fhadoop%2Fservices%2F.

Reply:
They need to be fixed. Google3 has warnings for them.

Reply:
`/usr/share/google/get_metadata_value` throws exit-code 22 if metadata key is not found. Maybe that's why in the example script, they are using it like this to suppress the error - https://github.com/GoogleCloudDataproc/custom-images/blob/master/examples/customization_script.sh#L19

Reply:
No, the error will not be suppressed. Also don't paste Google internal links to Github, remove it please.

Reply:
I have removed the link, thanks for pointing it out. As mentioned above, `/usr/share/google/get_metadata_value` throws exit-code 22 if metadata key is not found. If I separate definition and initialization of my variables, my script fails because of that. I am following the example given in the repo - https://github.com/GoogleCloudDataproc/custom-images/blob/master/examples/customization_script.sh#L19, and it works as intended. We might need to fix implementation of function and the example script, but let's do that separately.